subprojects/core-api/src/main/java/org/gradle/api/artifacts/transform/ArtifactTransformAction.java

```diff
@@ -25,11 +25,11 @@
  *
  * <p>Implementations can receive parameters by using annotated abstract getter methods.</p>
  *
- * <p>A property annotated with {@link TransformParameters} will receive the object provided by {@link ParameterizedArtifactTransformSpec#getParameters()}.
+ * <p>A property annotated with {@link ArtifactTransformParameters} will receive the object provided by {@link ParameterizedArtifactTransformSpec#getParameters()}.
```

Comment:
So we are standardizing on `ArtifactTransform` and not `Transform`, as the prefix? I thought we said we'd do otherwise. Is there a strong reason for this? If not, I'd go with the simpler thing.

Reply:
My understanding was that we would use the most specific thing, and that is `ArtifactTransform`. A `Transform` would be something which is not necessarily limited to dependency management nor files.

This was the relevant discussion:
> * _Artifact_ transforms? That is work that takes dependency artefacts and produces more dependency artefacts. Do we want to make these APIs specific to artifact transforms only?
> * _File_ transforms? That is, work that takes files (which a dependency artefact happens to be) and produces more files (which happen to be used as dependency artefacts right now). Do we want to define APIs that could be used for any work that takes files and produces files?
> * _transforms_? That is, work that takes some parameters (which may or may not include files) and produces some output (which may or may not include files)? Do we want to define APIs that could be used for any kind of transformation?
> * _work actions_? That is, work that may have some parameters and may produce an output.

Reply:
I agree we should go with "artifact transforms". What I'm suggesting is that we should consider the FQCN instead of the simple name of the types.

Reply:
I don't think the package name is very visible. If it would, we could directly go for no prefix at all and use `Action`, `Outputs`, etc., given that the package name is `artifacts.transform`.

I am not very opposed to using `Transform` as a common prefix. We cannot rename legacy things like `ArtifactTransformException`, though. We can use the usual deprecation though.

That would also mean that we have `TransformSpec` and `ParameterizedTransformSpec` for the registration, right?

Reply:
I changed everything to `Transform...`.